.github/workflows/testing.yml

```diff
@@ -0,0 +1,42 @@
+name: Unit Tests
+
+on:
+  push:
+  pull_request:
+  schedule:
+      - cron: '00 4 * * *'  # daily at 4AM
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+      fail-fast: false
+    steps:
+
+    - uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install
+      shell: bash -l {0}
+      run: source continuous_integration/scripts/install.sh
+
+    - name: Install test requirements
+      shell: bash -l {0}
+      run: |
+        set -vxeuo pipefail
+        python -m pip install -r requirements-dev.txt
+        python -m pip list
+
+    - name: Test with pytest
+      shell: bash -l {0}
+      run: |
+        set -vxeuo pipefail
+        coverage run -m pytest -v
```

Comment:
Do we still care about `coverage`? I thought it had a security issue, and people started abandoning it...

Reply:
We weren't exposed to the security issue (which was specific to the bash uploader).

It's reasonable to discuss whether we want to continue to use coverage, but we currently do so in most (all?) repos --- see [bluesky/bluesky](https://github.com/bluesky/bluesky/blob/ebffcc55c7d28e279ce101cb79ea1ff4a3d92411/.github/workflows/testing.yml#L59-L60) --- so I think we should re-evaluate in a separate conversation and stay consistent for now.